app/controllers/forumthreads_controller.rb

```diff
@@ -80,6 +85,20 @@ def destroy
     redirect_to @thread.forum
   end
 
+  def search
+  end
+
+  def search_redirect
```

Comment:
I've tried everything I could think of. There are two options:
1. Everything is moved to `index` and after and/or during the changing of params, the code redirects back to index to clear the params out of the URL.
2. The form points to `search_redirect` which changes all of the params at once and then redirects to `index` at the end.
I personally prefer the second one.